docs/contributing/component-js-configuration-options.md

```diff
@@ -0,0 +1,192 @@
+
+`data-*` attributes can be found in JavaScript by looking at an element's `dataset` property. Browsers will helpfully convert the attribute names from HTML's kebab-case to more JavaScript-friendly camelCase when working with `dataset`. See ['Naming configuration options'](#naming-configuration-options) for caveats.
+
+As we expect configuration-related `data-*` attributes to always be on the component's root element (the same element with the `data-module` attribute), we can access them all quickly using `$module.dataset`.
+
+Let's update our `mergeConfigs` call from earlier to include `$module.dataset` as the highest priority.
+
+```javascript
+import { mergeConfigs } from '../../common/index.mjs'
+import { normaliseDataset } from '../../common/normalise-dataset.mjs'
+
+this.config = mergeConfigs(
+  defaultConfig,
+  config || {},
+  normaliseDataset($module.dataset)
+)
+```
+
+Here, we pass it through our `normaliseDataset` function. This is because attribute values in dataset are always interpreted as being strings. `normaliseDataset` runs a few simple checks to convert values back to numbers or booleans where appropriate.
+
+Now, in our HTML, we could pass configuration options by using the kebab-case version of the option's name.
+
+```html
+<div
+  class="govuk-accordion"
+  data-module="govuk-accordion"
+  data-remember-expanded="false">
+  ...
+</div>
+```
+
+However, this only works for developers who are writing raw HTML. We include Nunjucks macros for each component with GOV.UK Frontend to make development easier and faster, but this also makes it harder for developers to manually alter the raw HTML. We'll need to add a new parameter to Nunjucks to help them out.
+
+## Adding a Nunjucks parameter
+
+Although most components support adding arbitrary attributes and values via the `attributes` parameter, not all of them do. This method is also more verbose compared to having a dedicated Nunjucks parameter.
+
+Let's update our previous bit of HTML to make the `data-*` attribute changeable.
+
+```nunjucks
+<div
+  class="govuk-accordion"
+  data-module="govuk-accordion"
+  {%- if params.rememberExpanded %} data-remember-expanded="{{ params.rememberExpanded | escape }}"{% endif %}>
+  ...
+</div>
+```
+
+The above checks for the existence of the `rememberExpanded` parameter and adds the `data-remember-expanded` attribute if it's present. It then passes in the developer's defined value, running it through [Nunjucks' native `escape` filter](https://mozilla.github.io/nunjucks/templating.html#escape-aliased-as-e) to ensure that values can't break our HTML.
+
+We can now call the Accordion's Nunjucks macro with our new `rememberExpanded` parameter:
+
+```nunjucks
+{% from "govuk/components/accordion/macro.njk" import govukAccordion %}
+
+{{ govukAccordion({
+  id: "accordion",
+  rememberExpanded: false,
+  items: [...]
+}) }}
+```
+
+And that's pretty much it!
```

Comment:
Although I want this line to stay because it's cheery, I think there needs to be a 'wrap up' line. Something about the benefit of using this method perhaps?